boards/common/stm32/Kconfig.l4

```diff
@@ -0,0 +1,198 @@
+# Copyright (c) 2020 Inria
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+menu "STM32 L4/WB clock configuration"
+    depends on CPU_FAM_L4 || CPU_FAM_WB
+
+choice
+bool "Clock source selection"
+default USE_CLOCK_PLL
+
+config USE_CLOCK_PLL
+    bool "Select PLL as system clock"
+
+config USE_CLOCK_MSI
+    bool "Select MSI as system clock"
+
+config USE_CLOCK_HSE
+    bool "Select HSE (High Speed External) oscillator as system clock" if BOARD_HAS_HSE
+
+config USE_CLOCK_HSI
+    bool "Select HSI16 (High Speed Internal) oscillator as system clock"
+
+endchoice
+
+choice
+bool "Source clock for PLL" if USE_CLOCK_PLL
+default CLOCK_PLL_SRC_MSI if !BOARD_HAS_HSE
+default CLOCK_PLL_SRC_HSE if BOARD_HAS_HSE
+
+config CLOCK_PLL_SRC_MSI
+    bool "Use MSI source clock"
+
+config CLOCK_PLL_SRC_HSE
+    bool "Use HSE source clock"
+    depends on BOARD_HAS_HSE
+
+config CLOCK_PLL_SRC_HSI
+    bool "Use HSI16 source clock"
+endchoice
+
+choice
+bool "Desired MSI clock frequency" if USE_CLOCK_MSI || (USE_CLOCK_PLL && CLOCK_PLL_SRC_MSI)
+default CLOCK_MSI_48MHZ
+
+config CLOCK_MSI_100KHZ
+    bool "100kHz"
+
+config CLOCK_MSI_200KHZ
+    bool "200kHz"
+
+config CLOCK_MSI_400KHZ
+    bool "400kHz"
+
+config CLOCK_MSI_800KHZ
+    bool "800kHz"
+
+config CLOCK_MSI_1MHZ
+    bool "1MHz"
+
+config CLOCK_MSI_2MHZ
+    bool "2MHz"
+
+config CLOCK_MSI_4MHZ
+    bool "4MHz"
+
+config CLOCK_MSI_8MHZ
+    bool "8MHz"
+
+config CLOCK_MSI_16MHZ
+    bool "16MHz"
+
+config CLOCK_MSI_24MHZ
+    bool "24MHz"
+
+config CLOCK_MSI_32MHZ
+    bool "32MHz"
+
+config CLOCK_MSI_48MHZ
+    bool "48MHz"
+
+endchoice
+
+config CLOCK_MSI
+    int
+    default 100000 if CLOCK_MSI_100KHZ
+    default 200000 if CLOCK_MSI_200KHZ
+    default 400000 if CLOCK_MSI_400KHZ
+    default 800000 if CLOCK_MSI_800KHZ
+    default 1000000 if CLOCK_MSI_1MHZ
+    default 2000000 if CLOCK_MSI_2MHZ
+    default 4000000 if CLOCK_MSI_4MHZ
+    default 8000000 if CLOCK_MSI_8MHZ
+    default 16000000 if CLOCK_MSI_16MHZ
+    default 24000000 if CLOCK_MSI_24MHZ
+    default 32000000 if CLOCK_MSI_32MHZ
+    default 48000000 if CLOCK_MSI_48MHZ
```

Comment:
If this value needs to have always a value, a unconditional default should be added.